src/Runner.Common/JobServerQueue.cs

```diff
@@ -93,6 +99,10 @@ public void Start(Pipelines.AgentJobRequestMessage jobRequest)
 
             _jobServer.InitializeWebsocketClient(serviceEndPoint);
 
+            VssCredentials serverCredential = VssUtil.GetVssCredential(serviceEndPoint);
+
+            _jobServer.InitializeResultsClient(new Uri(Environment.GetEnvironmentVariable("ResultsReceiverEndpoint")), serverCredential.ToOAuthCredentials());
```

Comment:
When testing this in a codespace we saw an error running a job because `Environment.GetEnvironmentVariable("ResultsReceiverEndpoint"))` was null. Which actually makes sense because I don't think that's supposed to be an environment variable 😄 so we need to figure out how we can get that value here when it is set by the actions service.

Also not super confident on the serverCredential here, either.